traits/ctraits.c

```diff
@@ -4070,10 +4090,32 @@ validate_trait_complex(
                 return result;
 
             case 22: /* Callable check: */
-                if (value == Py_None || PyCallable_Check(value)) {
-                    goto done;
+                {
+                    if (PyCallable_Check(value)) {
+                        return value;
+                    }
+                    else if (value == Py_None) {
+
+                        int tuple_size = PyTuple_GET_SIZE(trait->py_validate);
+
+                        //Handle callables without allow_none, default to allow None
+                        if(tuple_size < 2){
```

Comment:
Same comment as in previous code: let's keep spacing consistent with what's already there.
```suggestion
                        if (tuple_size < 2) {
```